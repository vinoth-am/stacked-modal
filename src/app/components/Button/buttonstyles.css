.button {
  font-size: 16px;
  font-weight: bold;
  width: 150px;
  height: 40px;
  color: var(--white);
  border-radius: 6px;
  border-color: transparent;

  background-color: var(--primary-blue);
}

.secondary {
  background-color: var(--secondary-red);
}
