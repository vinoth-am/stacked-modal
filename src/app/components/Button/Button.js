import React from "react";
import "./buttonstyles.css";

export const Button = (props) => {
  const { onClick, color, style } = props;

  return (
    <button
      data-testid="actionButton"
      style={style}
      className={`${"button"} ${color}`}
      onClick={(data) => onClick(data)}
    >
      {props.children}
    </button>
  );
};
