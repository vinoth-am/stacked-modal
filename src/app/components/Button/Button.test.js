import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Button } from "./Button";

describe("<Button />", () => {
  const onClick = jest.fn();
  it("Button snapshot", () => {
    const { button } = render(<Button onClick={onClick} />);
    expect(button).toMatchSnapshot();
  });

  it("Validate the button", () => {
    const { getByTestId } = render(<Button onClick={onClick} />);
    const actionButton = getByTestId("actionButton");
    expect(actionButton).toBeTruthy();
  });

  it("Validate the button text", () => {
    const { getByTestId } = render(<Button onClick={onClick}>Open</Button>);
    const actionButton = getByTestId("actionButton");
    expect(actionButton).toHaveTextContent("Open");
  });

  it("Validate the button click", () => {
    const { getByTestId } = render(<Button onClick={onClick} />);
    const actionButton = getByTestId("actionButton");
    fireEvent.click(actionButton);
    expect(onClick).toBeCalledTimes(1);
  });
});
