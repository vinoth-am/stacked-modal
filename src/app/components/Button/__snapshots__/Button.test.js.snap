// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button snapshot 1`] = `undefined`;
